### PR TITLE
Add "Update available" dialog

### PR DESCRIPTION
**Pull request by FabianDevel**
Depends on https://github.com/Infomaniak/android-core/pull/48

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/data/LocalSettings.kt [hunk 1]
```diff
@@ -41,6 +41,12 @@ class LocalSettings private constructor(context: Context) {
 
     fun removeSettings() = sharedPreferences.transaction { clear() }
 
+    //region App Launches Number
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/data/LocalSettings.kt:44]:
```suggestion
    //region App launches
```

File: app/src/main/java/com/infomaniak/mail/data/LocalSettings.kt [hunk 2]
```diff
@@ -76,6 +82,12 @@ class LocalSettings private constructor(context: Context) {
         set(value) = sharedPreferences.transaction { putBoolean(IS_APP_LOCKED_KEY, value) }
     //endregion
 
+    //region Update Later
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/data/LocalSettings.kt:85]:
```suggestion
    //region Update later
```

File: app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt [hunk 3]
```diff
@@ -53,6 +56,7 @@ class MainActivity : ThemedActivity() {
     private val mainViewModel: MainViewModel by viewModels()
 
     private val permissionUtils by lazy { PermissionUtils(this).also { registerMainPermissions(it) } }
+    private val localSettings by lazy { LocalSettings.getInstance(this) }
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt:59]:
```suggestion
    private val localSettings by lazy { LocalSettings.getInstance(this) }
    private val permissionUtils by lazy { PermissionUtils(this).also(::registerMainPermissions) }
```

File: app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt [hunk 4]
```diff
@@ -119,6 +125,7 @@ class MainActivity : ThemedActivity() {
 
     override fun onResume() {
         super.onResume()
+        localSettings.appLaunches++
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt:128]: Maybe move it to `onStart()` or `onCreate()` ?

Comment (FabianDevel) [reply app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt:128]: OnCreate may not be enough, as one can just put his app in background but often uses it so it is never killed by the system
But onStart seems better indeed

**Review by KevinBoulongne — APPROVED** [review]

**Review by JorisBodin — APPROVED** [review]

**Review by KevinBoulongne — APPROVED** [review]

**Pull request merged**